src/accounts.js

```diff
@@ -288,29 +289,66 @@ async function addPermission(authAccountName, keys, permissionName, parentPermis
     actions = actions.concat(linkActions);
   }
 
-  return this.transact(actions, broadcast);
+  if (confirm) {
```

Comment:
should ideally check a boolean like  === true to reduce the likelihood of false positives.